### ISSUE TITLE
Suggestion: better tracking chapter progress, through a browser extension 

### ISSUE THREAD

**Issue by yiransheng** [seanhess/serials#38]
Been trying out the app, pretty neat.

One thing occured to me (correct me if I am wrong), the current flow for bookmark/tracking progress is:
1. click a chapter
2. read
3. go back to webfiction app
4. click next chapter through UI

Many fictions have next/prev chapter links built in (wordpress ones for example), readers are much more likely to click on those instead of clicking back button - and the progress happened this way is lost. 

Two ways I can think of to address this:
1. through link's `:visited` style, however, in all browsers, js have no access to the data, only css does, so cannot do it automatically, still requires user to manually mark all visited chapters as read
2. also bundle a chrome extension with the app, might work like this:
- have a background page/script running to coordinate events across tabs
- when user visit webfiction app, somehow share authentication token with the background script, so it can communicate with app server on its own
- when user click open a chapter from app, inform the background script, indicating a new reading session has started, and keep tracking the opened fiction tab 
- run a content-script on page load in fiction/chapter tab, renders prev/next tab button, the content script should know the current chapter link and webfiction id, so are its previous and next chapters' link and id
- clicking these buttons will refresh page navigate to next chapter's link (can be just a `<a>` tag), also communicating to background page the current chapter is done
- upon receiving the message from content script, background page make a ajax request to webfiction server, with the chapter id provided in the message passed in

tl;dr: use browser extension to render prev/next button on actual fiction chapter's page and communicate to server about reading progress upon clicking them

**Comment by seanhess** [comment]
Hi thanks for the suggestion! I agree, this would be awesome. 

First, did you discover that you can mark a chapter as read by clicking the circle? The bookmark is set to the chapter after the last one that has been read. So you can get the bookmark to the correct place with one click if you know where you are. 

But yeah, I would love to make it better. I already have a way of solving this technically if I deliver the chapters from my server, but then I'm no longer "just an index", and I don't want the authors to worry.
1. Does 1 offer anything over the current functionality? Either you didn't notice the mark-as-read feature or I don't understand. 
2. This is a great idea, but I wonder how many people would install it? I think we have to be ok with whatever functionality exists on the site itself for the majority of users.

**Comment by seanhess** [comment]
Closing this for now, but I'd be open to discuss further. I'm still curious about your answers to my questions. 

I'll have to do some testing to see if users are ok with the flow as it is, or if we need to get more fancy.
